tests/system-containers/main.yml

```diff
@@ -0,0 +1,511 @@
+---
+# vim: set ft=ansible:
+#
+# !!!NOTE!!! This playbook was tested using Ansible 2.1; it is recommended
+# that the same version is used.
+#
+# Test Cases
+#   Core Functionality
+#     - Verify system containers can be installed through atomic command
+#     - Verify system containers can be uninstalled through the atomic command
+#     - Verify user can specify name for system containers
+#     - Verify system containers can be listed
+#     - Verify environment variables can be pass to the container
+#     - Verify update/rollback of system containers
+#     - Verify commands can be run in the system container
+#     - Verify specification of rootfs for system containers.
+#     - Verify setting RUN_DIRECTORY and STATE_DIRECTORY
+#     - Verify system containers are started on reboot
+#     - Verify system containers persist through reboot
+#
+#   Flannel & Etcd Tests
+#     - Verify installation of flannel and etcd containers
+#
+#   Negative Testing
+#     - Verify error message is displayed when uninstalling a system container that does not exist
+#
+#   Not Covered
+#     Upgrade Tests
+#       - Verify the system container persists through ostree upgrade
+#       - Verify the system container persists through ostree rollback
+#
+- name: System Containers - Setup
+  hosts: all
+  become: yes
+
+  tags:
+    - setup
+
+  vars_files:
+    - vars.yml
+
+  pre_tasks:
+    - name: Set facts
+      set_fact:
+        etcd_image: 'registry.access.redhat.com/rhel7/etcd'
+        etcd_name: 'etcd'
+        flannel_image: 'registry.access.redhat.com/rhel7/flannel'
+        flannel_name: 'flannel'
+      when: ansible_distribution == 'RedHat'
+
+    - name: Set facts
+      set_fact:
+        etcd_image: 'gscrivano/etcd'
+        etcd_name: 'etcd'
+        flannel_image: 'gscrivano/flannel'
+        flannel_name: 'flannel'
+      when: ansible_distribution != 'RedHat'
```

Comment:
It looks like we may have proper upstream images soon.

https://lists.projectatomic.io/projectatomic-archives/atomic-devel/2017-January/msg00024.html

When they land, we'll want to update this playbook to use them.